maintainer-guide.md

```diff
@@ -174,6 +174,11 @@ An issue for data migration should have been created after the breaking changes
   * While this may be a suitable beginner-level issue, in the interest of keeping as few legacy code in the code base as possible, it should be done by a team member with a minimum delay.
 * Close the data migration issue.
 
+**Tips for data migration**
```

Comment:
This doesn't really belong in the maintainer's guide. Let's create a new document `best-practices.md` to contain this kind of stuff.